### PR TITLE
IDEController Draft

### PR DESCRIPTION
**Pull request by vphilippon**
IDEController draft, which could be used for test purpose.
I need to read more about concurency with cherrypy/python and config before going foward.
@grondinjc Check please.

### CODE REVIEW

**Comment by vphilippon** [comment]
@grondinjc @maym2104 J'ai compléter la structure de config.
J'ai changer les .conf en .conf.example.
Vous pourrez vous faire des copies en enlevant le .example et en mettant les infos propres à vos machines.
L'un de vous 2 peux faire un review et merger si tout est bon!

File: bin/configs/dummy_controller.conf.example [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+SRC = "/srv/www/cide.py/src"
+
+# -- Static content config --
+# Set root (/) of static dir/file to src
+# Just a shortcut for further config. Does not give static access to src folder
+[/]
+tools.staticdir.root = %(SRC)s
+tools.staticfile.root = %(SRC)s
+
+# Maps domain.com/<<DUMMT>>/static to src/static
```
Comment (grondinjc) [inline bin/configs/dummy_controller.conf.example]: @vphilippon did you meant "dummy" ?

File: src/server/welcomeController.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class WelcomeController(object):
+  """
+  Empty Controller to distrubute static welcoming content.
```
Comment (grondinjc) [inline src/server/welcomeController.py:3]: :+1:

**Comment by grondinjc** [comment]
@vphilippon This is good but I would add something..

I would add a python (or bash) script that would somehow take a dictionnary as follow :
Key : "PLACEHOLDER_STR",Value: "VALUE"

The script would create the config files based on the examples and would replace all
detected placeholders by the assigned values ... This would allow to change all configuration file
SRC placeholder quickly

File: bin/cide.conf.example [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[DEFAULT]
+server = ./configs/server.conf
+welcomeController = ./configs/welcomeController.conf
+ideController = ./configs/ideController.conf
```
Comment (grondinjc) [inline bin/cide.conf.example:4]: wrap values in quotes ("./configs/ideController.conf") for uniformity ?

**Comment by grondinjc** [comment]
waiting for approuval.

File: bin/cide.conf.example (hunk 3 above)
Comment (vphilippon) [inline bin/cide.conf.example:4]: If I quote them, the quotes get escaped and the os.path.join() returns something like :
`bin/'./configs/server.conf'`.
It bugged me, but I didn't find a way to fix it.
EDIT: Googled a bit. That's a "feature" of Python 2.6+. Yay.

File: bin/configs/dummy_controller.conf.example (hunk 1 above)
Comment (vphilippon) [inline bin/configs/dummy_controller.conf.example]: @grondinjc Indeed, will fix in a minute.

**Comment by vphilippon** [comment]
@grondinjc The script to create config files is an interesting idea. Although, I would still merge the PR (after the 'DUMMT' correction) and take a look at this afterward.

**Comment by vphilippon** [comment]
@grondinjc @maym2104 Correction made. Merge/comment as you see fit.
Thanks!

File: bin/cide.conf.example (hunk 3 above)
Comment (grondinjc) [inline bin/cide.conf.example:4]: ok then

**Pull request merged**